### ISSUE TITLE
Missing tests for Char.flix and error in Char.isHexDigit

### ISSUE THREAD

**Issue by stephentetley** [flix/flix#1036]
Hi Magnus, I noticed `Char.flix` is missing a test module so I wrote one written one while doing the String additions.

The test module is currently on a branch here:

https://github.com/stephentetley/flix/blob/string-additions2/main/test/ca/uwaterloo/flix/library/TestChar.flix

The tests have show that isHexDigit isn't working properly - it fails for `Char.isHexDigit('8')` and `Char.isHexDigit('9')`.

Rather than change `flix.runtime.library.Char` where the primitive is defined I'd suggest we add these two functions to `Char.flix`:

~~~

///
/// Returns the character `c` as an Int32.
///
pub def toInt32(c: Char): Int32 =
    c as Int32

///
/// Returns the respective character for the int `i`.
///
pub def fromInt32(i: Int32): Char =
    i as Char

~~~

Then we can have an easy definition for `isHexDigit` in Flix:

~~~

///
/// Returns `true` if the given char `c` is in the range 0...F.
///
pub def isHexDigit(c: Char): Bool = match toInt32(c) {
    case i if i >= 0x0030 && i <= 0x0039 => true        // '0'..'9'
    case i if i >= 0x0041 && i <= 0x0046 => true        // 'A'..'F'
    case i if i >= 0x0061 && i <= 0x0066 => true        // 'a'..'f'
    case _ => false
}

~~~

We can also have similar definitions in Flix for `isAscii` and `isOctDigit` removing the dependency on `flix.runtime.library.Char`.

What do you think?

Thanks

**Comment by magnus-madsen** [comment]
Sounds good. 

I prefer anything that reduces the dependence on Java classes.

**Comment by stephentetley** [comment]
Thanks - I'll redo `isAscii` and `isOctDigit` and put it on a new separate branch so it can be it's own pull request.

**Comment by stephentetley** [comment]
Shall I add a function `isAsciiDigit` just to test for the characters between '0' and '9'?

The current `isDigit` also includes other ranges, see link:

https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/lang/Character.html#isDigit(char)

**Comment by magnus-madsen** [comment]
> 
> 
> Shall I add a function `isAsciiDigit` just to test for the characters between '0' and '9'?
> 
> The current `isDigit` also includes other character ranges, see link:
> 
> https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/lang/Character.html#isDigit(char)

Gosh! That is some surprising behavior. I think an `isAsciiDigit` is entirely reasonable.

**Comment by stephentetley** [comment]
Thanks - I will do.

**Comment by magnus-madsen** [comment]
Fixed in #1036
